frontend/src/result.js

```diff
@@ -32,7 +32,13 @@ export class Result extends React.Component {
       return;
     }
     fetch(Settings.serverUrl + '/result/' + this.state.id)
-      .then(response => response.json())
+      .then(response => {
+        this.setState({ "httpStatus": response.status });
+        if (!response.ok) {
+          throw new Error("Failed with HTTP code " + response.status);
+        }
+        return response.json();
+      })
```

Comment:
Rather than having the `httpStatus` be a state variable, I think we should make `isResultValid` (or `isNotFound`) a state variable and update it here.

That way we can avoid the conditional in the `render` function below.